homeassistant/components/airtouch5/cover.py

```diff
@@ -0,0 +1,139 @@
+"""Representation of the Damper for AirTouch 5 Devices."""
+import logging
+from typing import Any
+
+from airtouch5py.airtouch5_simple_client import Airtouch5SimpleClient
+from airtouch5py.packets.zone_control import (
+    ZoneControlZone,
+    ZoneSettingPower,
+    ZoneSettingValue,
+)
+from airtouch5py.packets.zone_name import ZoneName
+from airtouch5py.packets.zone_status import ZoneStatusZone
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityFeature,
+)
+from . import Airtouch5ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import Airtouch5Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: Airtouch5ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Airtouch 5 Cover entities."""
+    client = config_entry.runtime_data
+
+    entities: list[CoverEntity] = []
+
+    # Each zone has a cover for its open percentage
+    for zone in client.zones:
+        entities.append(
+            Airtouch5ZoneOpenPercentage(
+                client, zone, client.latest_zone_status[zone.zone_number].has_sensor
+            )
+        )
+
+    async_add_entities(entities)
+
+
+class Airtouch5ZoneOpenPercentage(CoverEntity, Airtouch5Entity):
+    """How open the damper is in each zone."""
+
+    _attr_device_class = CoverDeviceClass.DAMPER
+
+    def __init__(
+        self, client: Airtouch5SimpleClient, name: ZoneName, has_sensor: bool
+    ) -> None:
+        """Initialise the Cover Entity."""
+        super().__init__(client)
+        self._name = name
```

Comment:
What's this for?

Reply:
Probably not needed. I'll remove and re-test.

Reply:
Name is required. Fails to initialize if I remove it.

Reply:
It's used in the callbacks of this class.

https://github.com/home-assistant/core/pull/122462/files#diff-85d5bede0b4d8818017cc2b712764f6b4e672c37057d40897cecb9d3896371b0R81-R84

Reply:
Oh can we rename this to `zone_name`? `_name` kinda indicates that's it's the entity name

Reply:
done